contracts/babylon/src/utils/btc_light_client_querier.rs

```diff
@@ -0,0 +1,31 @@
+use crate::error::ContractError;
+use crate::state::config::CONFIG;
+use btc_light_client::msg::btc_header::BtcHeaderResponse;
+use cosmwasm_std::Deps;
+
+/// Get the BTC light client contract address
+fn get_contract_addr(deps: Deps) -> Result<String, ContractError> {
+    let cfg = CONFIG.load(deps.storage)?;
+    Ok(cfg
+        .btc_light_client
+        .ok_or(ContractError::BtcLightClientNotSet {})?
+        .to_string())
+}
+
+/// Query the tip header from the BTC light client
+pub fn query_tip_header(deps: Deps) -> Result<BtcHeaderResponse, ContractError> {
+    let contract_addr = get_contract_addr(deps)?;
+    let msg = btc_light_client::msg::contract::QueryMsg::BtcTipHeader {};
+    let response: BtcHeaderResponse = deps.querier.query_wasm_smart(&contract_addr, &msg)?;
+    Ok(response)
+}
+
+/// Query a header by hash from the BTC light client
+pub fn query_header_by_hash(deps: Deps, hash: &str) -> Result<BtcHeaderResponse, ContractError> {
+    let contract_addr = get_contract_addr(deps)?;
+    let msg = btc_light_client::msg::contract::QueryMsg::BtcHeaderByHash {
+        hash: hash.to_string(),
+    };
+    let response: BtcHeaderResponse = deps.querier.query_wasm_smart(&contract_addr, &msg)?;
+    Ok(response)
+}
```

Comment:
I guess as long as we keep the checkpointing and epoching funtionality in the Babylon contract (which makes sense btw), these nested queries make sense as well.